src/lib/support/tests/TestBufferWriter.cpp

```diff
@@ -77,6 +77,29 @@ class BWTest : public Base
 
 using namespace chip::Encoding;
 
+void TestSpanVersusRegular(nlTestSuite * inSuite, void * inContext)
+{
+    uint8_t buf_regular[5]    = { 0, 0, 0, 0, 0 };
+    uint8_t buf_span[5]       = { 0, 0, 0, 0, 0 };
+    uint8_t all_zeroes[5]     = { 0, 0, 0, 0, 0 };
+    uint8_t final_expected[5] = { 1, 2, 3, 4, 0 };
+
+    BufferWriter regular_writer(buf_regular, sizeof(buf_regular));
+    BufferWriter span_writer(chip::MutableByteSpan{ buf_span, sizeof(buf_span) });
```

Comment:
```suggestion
    BufferWriter span_writer(chip::MutableByteSpan{ buf_span });
```
Span has a constructor from array literal that does the right thing.